validator/project_validator.go

```diff
@@ -1021,16 +1016,25 @@ func verifyTaskDependencies(project *model.Project) ValidationErrors {
 			}
 
 			// check that name of the dependency task is valid
-			if dep.Name != model.AllDependencies && !taskNames[dep.Name] {
+			if dep.Name != model.AllDependencies && project.FindProjectTask(dep.Name) == nil {
 				errs = append(errs,
 					ValidationError{
+						Level: Warning,
 						Message: fmt.Sprintf("project '%s' contains a "+
 							"non-existent task name '%s' in dependencies for "+
 							"task '%s'", project.Identifier, dep.Name,
 							task.Name),
 					},
 				)
 			}
+			if dep.Variant != "" && dep.Variant != model.AllVariants && project.FindBuildVariant(dep.Variant) == nil {
+				errs = append(errs, ValidationError{
+					Level: Warning,
+					Message: fmt.Sprintf("project '%s' contains a dependency '%s' on nonexistent buildvariant '%s'",
```

Comment:
I think this should be worded how the above is: "contains a non-existent variant name <dep.Variant>  in dependencies for task <task.Name>"